### ISSUE TITLE
MuMillipede should not add track segments to the hypothesis without a configured MuonPhotonicsService (Trac #1968)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1960]
A lazy user like me might assume that setting MuonPhotonicsService=None in MuMillipede means that it does not add track segments to the hypothesis. However, this also has to be explicitly turned off by setting MuonSpacing=0. Not knowing this causes weird behavior of the energy losses if the type of the seed track is an EMinus, i.e. half of the energy will be saved as Cascades and half as ContainedTracks.

This is not desirable. Setting EITHER MuonPhotonicsService=None OR MuonSpacing=0 should be enough to tell millipede not to add track segments to the hypotheses and not to add muon losses. The same goes for ShowerSpacing and CascadePhotonicsService. 

Migrated from https://code.icecube.wisc.edu/ticket/1968
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:13:57", 
    "description": "A lazy user like me might assume that setting MuonPhotonicsService=None in MuMillipede means that it does not add track segments to the hypothesis. However, this also has to be explicitly turned off by setting MuonSpacing=0. Not knowing this causes weird behavior of the energy losses if the type of the seed track is an EMinus, i.e. half of the energy will be saved as Cascades and half as ContainedTracks.\n\nThis is not desirable. Setting EITHER MuonPhotonicsService=None OR MuonSpacing=0 should be enough to tell millipede not to add track segments to the hypotheses and not to add muon losses. The same goes for ShowerSpacing and CascadePhotonicsService. ", 
    "reporter": "musner", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067237750774", 
    "component": "combo reconstruction", 
    "summary": "MuMillipede should not add track segments to the hypothesis without a configured MuonPhotonicsService", 
    "priority": "normal", 
    "keywords": "millipede", 
    "time": "2017-03-21T15:10:56", 
    "milestone": "", 
    "owner": "jbraun", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20170321T15:11:49: musner changed summary from "MuMillipede should not add track segments to the hypothesis without a configued Muon PhotonicsService" to "MuMillipede should not add track segments to the hypothesis without a configured MuonPhotonicsService"

**Comment by icecube-trac** [comment]
Trac update at 20170321T18:24:59: 
* olivas changed status from "new" to "assigned"
* olivas changed owner from "" to "jbraun"

**Comment by icecube-trac** [comment]
Trac update at 20170321T18:30:03: 
* jbraun changed status from "assigned" to "accepted"
* jbraun changed milestone from "" to "May 2017"

**Comment by icecube-trac** [comment]
Trac update at 20170517T16:34:43: jbraun changed milestone from "May 2017" to "June 2017"

**Comment by icecube-trac** [comment]
Trac update at 20170623T19:19:59: jbraun commented:

> Agreed, but the correct behavior is to log_fatal() if muon segments are requested but no MuonPhotonicsService is given.

**Comment by icecube-trac** [comment]
Trac update at 20170623T19:35:24: 
* jbraun changed resolution from "" to "fixed"
* jbraun commented:

> In changeset 156726:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="156726"
> Require user to explicitly set muon/cascade spacing to zero if no Muon/Cascade PhotonicsService is provided.  Closes #1960.
> ```


* jbraun changed status from "accepted" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:13:57: 
* olivas commented:

> Milestone June 2017 deleted


* olivas changed milestone from "June 2017" to ""
